File: javaSpringShip/src/main/java/pl/battleships/javaspringship/client/SimpleShotHandler.java
package pl.battleships.javaspringship.client;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.client.RestTemplate;
import pl.battleships.core.api.ShotHandler;
import pl.battleships.core.model.Position;
import pl.battleships.core.model.ShotResult;
import pl.battleships.javaspringship.config.GameProperties;

import java.util.Map;

@Slf4j
@RequiredArgsConstructor
public class SimpleShotHandler implements ShotHandler {

    private final GameProperties properties;
    private final RestTemplate restTemplate;

    @Override
    public ShotResult shotToOpponent(String gameId, Position position) {
        log.debug("Sleeping {}ms and making shot to {} with position {}", properties.getSleep(),properties.getOpponentUrl(), position);
        try {
            Thread.sleep(properties.getSleep());
        } catch (InterruptedException e) {
            log.error("Problem while sleeping", e);
        }
        return restTemplate.postForEntity(properties.getOpponentUrl() + "/game/{id}/shot", position, ShotResult.class, Map.of("id", gameId))
                .getBody();
    }
}

File: kotlinSpringShip/src/main/kotlin/pl/battleships/kotlinspringship/client/SimpleShotHandler.kt
package pl.battleships.kotlinspringship.client

import mu.KotlinLogging
import org.springframework.web.client.RestTemplate
import pl.battleships.core.api.ShotHandler
import pl.battleships.core.model.Position
import pl.battleships.core.model.ShotResult
import pl.battleships.kotlinspringship.config.GameProperties
import java.util.Map

class SimpleShotHandler (
    private val restTemplate : RestTemplate,
    private val gameProperties: GameProperties
) : ShotHandler {
    private val log = KotlinLogging.logger {}

    override fun shotToOpponent(gameId: String, position: Position): ShotResult {
        log.info { "Sleeping ${gameProperties.sleep}ms and making shot to ${gameProperties.opponentUrl} with position $position" }
        try {
            Thread.sleep(gameProperties.sleep)
        } catch (e: InterruptedException) {
            log.error("Problem while sleeping", e)
        }
        return restTemplate
            .postForEntity(gameProperties.opponentUrl + "/game/{id}/shot", position, ShotResult::class.java, Map.of("id", gameId))
            .body!!
    }
}